tests/js/articles.frontend.spec.js

```diff
@@ -0,0 +1,139 @@
+} from '@wordpress/e2e-test-utils';
+
+import {
+  selectBlockByName,
+  openSidebarPanelWithTitle,
+  clickElementByText,
+  typeInDropdownWithLabel,
+} from './e2e-tests-helpers';
+
+import 'expect-puppeteer';
+
+const ARTICLS_BLOCK = `
+<!-- wp:planet4-blocks/articles {"article_count":2} -->
+<div class="wp-block-planet4-blocks-articles" data-render="planet4-blocks/articles" data-attributes="{&quot;attributes&quot;:{&quot;article_heading&quot;:&quot;Related Articles&quot;,&quot;article_count&quot;:2,&quot;tags&quot;:[],&quot;posts&quot;:[],&quot;post_types&quot;:[],&quot;read_more_text&quot;:&quot;Load More&quot;,&quot;read_more_link&quot;:&quot;&quot;,&quot;button_link_new_tab&quot;:false,&quot;ignore_categories&quot;:false},&quot;innerBlocks&quot;:[]}"></div>
+<!-- /wp:planet4-blocks/articles -->
+`;
+
+describe( 'Articles block frontend', () => {
+  beforeAll( async () => {
+    // This helps in overriding Wordpress dialogs preventing actions.
+    await enablePageDialogAccept();
+  } );
+
+  it ( 'test post thumbnail behaviour', async () => {
+    // Create new post.
+    await createNewPost({
+      title: 'Test Articles with no thumbnail',
+      content: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
+      showWelcomeGuide: false,
+    });
+    await publishPost();
+    await page.waitForSelector( '.post-publish-panel__postpublish' );
+
+    //Create a new page and add newly created post in it.
+    await createNewPost({
+      postType: 'page',
+      title: 'Test Articles block thumbnail',
+      showWelcomeGuide: false,
+    });
+
+    // Add articles block in page.
+    await setPostContent( ARTICLS_BLOCK );
+
+    await selectBlockByName( 'planet4-blocks/articles' );
+
+    await openSidebarPanelWithTitle( 'Setting' );
+
+    // Manually add post(without thumbnail img) in articles blocks.
+    await typeInDropdownWithLabel( 'Manual override', 'Test Articles with no thumbnail' );
+
+    // Manually add post(post with image inside post content, which was set as a feature image
+    // & display as a thumbnail in articles block) in articles blocks.
+    await typeInDropdownWithLabel( 'Manual override', 'Lorem ipsum dolor sit amet' );
+
+    // Publish the page.
+    await publishPost();
+
+    await new Promise((r) => setTimeout(r, 500));
+
+    await page.waitForSelector( '.post-publish-panel__postpublish' );
+
+    // Test articles block on frontend.
+    await clickElementByText( '*[@class="editor-post-publish-panel"]//a', 'View Page' );
+
+    await page.waitForNavigation();
+
+    await page.waitForSelector( '.page-template' );
+
+    // The "Articles Block" should appear on page.
+    await expect( page ).toMatchElement( '.block.articles-block' );
+
+    // If there is no image in the description of the post(also no featured image set),
+    // the blank space display as a thumbnail photo in articles block.
+    await expect(page).toMatchElement('article:nth-child(1) img', { src: ''} );
```

Comment:
Here again, since these articles were added using the manual override, the order might now have changed due to [planet-5462](https://jira.greenpeace.org/browse/PLANET-5462) and this might fail 😕

Reply:
We create a new post "Test Articles with no thumbnail" which will display first and the existing post "Lorem ipsum dolor sit amet" on second position, (as per new manual override rule the newly publish post will display first) hence the test will pass here :)